CMakeLists.txt

```diff
@@ -69,6 +69,9 @@ macro(build_mimick)
     ${cmake_commands}
     CMAKE_ARGS
       ${cmake_configure_args}
+    PATCH_COMMAND
+      sed -i "s/add_subdirectory \(test\)/ /g" CMakeLists.txt &&
+      sed -i "s/add_subdirectory \(sample\)/ /g" CMakeLists.txt
```

Comment:
This is almost certainly not going to work on Windows; we don't have `sed` there.  What we do elsewhere is to check a patch into the repository, and then use `git apply` as the PATCH_COMMAND; see https://github.com/ros2/libyaml_vendor/blob/9e14d58ce2198933f97a91adf2af31d788ae59aa/CMakeLists.txt#L91-L93, for instance.

Reply:
> Ah, good point. In that case, perhaps it would make sense to wrap this in an `if (BUILD_TESTING)`?

That won't help, because `BUILD_TESTING` is on by default on Windows anyway.

> And then, maybe it's better to just push the change into the source, given that it's been forked by Open Robotics anyway:

But good point; we should probably just patch this in the sources, since we are modifying them anyway.  Would you like to submit a PR to do that?

Reply:
Right yeah, I meant the BUILD_TESTING check would substitute for the `sed` operation in terms of allowing me to set the flag and disable this part of the build.